bombastic/api/src/server.rs

```diff
@@ -142,18 +146,16 @@ async fn query_sbom(
     let storage = &state.storage;
     // determine the encoding of the stored object, if any
     let encoding = storage.get_head(path.clone()).await.ok().and_then(|head| {
-        head.content_encoding.and_then(|ref e| {
-            accept_encoding
-                .negotiate([e.parse().unwrap()].iter())
-                .map(|s| s.to_string())
-                .filter(|x| x == e)
-        })
+        head.content_encoding
+            .as_ref()
+            .and_then(|e| e.parse::<Encoding>().ok())
+            .and_then(|e| accept_encoding.negotiate([&e].into_iter()).filter(|x| x == &e))
```

Comment:
I *DO* love this. I would've loved having this feedback when the PR was reviewed.